src/charm.py

```diff
@@ -1,34 +1,146 @@
 
-    def _on_install(self, _):
-        """Install prerequisites for the application"""
+    def _on_install(self, _: ops.charm.InstallEvent) -> None:
+        """Handle the install event (fired on startup)
+        Handles the startup install event -- installs updates the apt cache, installs mongoDB
+        Args:
+            _ (ops.charm.InstallEvent): The install event that fired.
+        Returns:
+            None: None
+        """
         self.unit.status = MaintenanceStatus("installing MongoDB")
         self._add_mongodb_org_repository()
         self._install_apt_packages(["mongodb-org"])
-        self.unit.status = WaitingStatus("Waiting to start MongoDB")
+        try:
+            mongo_path = subprocess.check_output(["which", "mongod"]).decode().strip()
+            if not mongo_path == "/usr/bin/mongod":
+                logger.error("wrong mongodb path found %s want /usr/bin/mongod", mongo_path)
+                self.unit.status = BlockedStatus("wrong mongod path")
+                return
+        except subprocess.CalledProcessError as e:
+            logger.error("failed to verify mongod installation %s", e)
+            self.unit.status = BlockedStatus("failed to verify mongod installation")
+            return
+
+    def _on_config_changed(self, _: ops.charm.ConfigChangedEvent) -> None:
+        """Event handler for configuration changed events.
+        Args:
+            _ (ops.charm.ConfigChangedEvent): The configuration changed event
+        Returns:
+            None: None
+        """
+        # TODO
+        # - update existing mongo configurations based on user preferences
+        # - add additional configurations as according to spec doc
+        with open("/etc/mongod.conf") as mongo_config_file:
+            mongo_config = yaml.safe_load(mongo_config_file)
+
+        machineIP = str(self.model.get_binding(PEER).network.bind_address)
+        bindIPs = "localhost," + machineIP
+        mongo_config["net"]["bindIp"] = bindIPs
+        if "replication" not in mongo_config:
+            mongo_config["replication"] = {}
+
+        mongo_config["replication"]["replSetName"] = "rs0"
+        with open("/etc/mongod.conf", "w") as mongo_config_file:
+            yaml.dump(mongo_config, mongo_config_file)
+
+        self.unit.status = ops.model.ActiveStatus()
+
+    def _on_start(self, event: ops.charm.StartEvent) -> None:
+        """Enables MongoDB service and initializes replica set
+        Args:
+            event (ops.charm.StartEvent): The triggering start event 
+        Returns:
+            None: None
+        """
+        logger.debug("Running on_start")
+        if not self.unit.is_leader():
+            return
+
+        # start mongo service
+        self._open_port_tcp(self.port)
+        self.unit.status = MaintenanceStatus("enabling MongoDB")
+        try:
+            logger.debug("enabling mongodb")
+            service_resume("mongod.service")
+            service_restart("mongod.service")
```

Comment:
I think you might be starting mongo, then restarting it. If you look at the [systemd lib](https://github.com/canonical/operator-libs-linux/blob/c9ed8d55b52adcc539ad0da8d604f91e29195ac9/lib/charms/operator_libs_linux/v0/systemd.py#L180), the `service_resume` function passes `--now` to systemctl, which means it'll get enabled *and* started :)

Reply:
Hi Jon! Yes, that is correct that I am starting mongod and then restarting it. I was following my script that I wrote for running MongoDB in which I do: `sudo systemctl enable mongod.service; sudo systemctl restart mongod.service` which I got from [this tutorial](https://computingforgeeks.com/how-to-setup-mongodb-replication-on-ubuntu/). But you are right that in the `systemd lib` they pass the `--now` flag and so I should be able to remove `service_restart` here